File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3036_number-of-subarrays-that-match-a-pattern-ii/solutions/python3/solution.py
class Solution:
    def countMatchingSubarrays(self, nums: list[int], pattern: list[int]) -> int:
        n = len(nums)
        m = len(pattern)
        subarray_size = m + 1
        count = 0
        
        for i in range(n - subarray_size + 1):
            match = True
            for k in range(m):
                if (pattern[k] == 1 and not (nums[i + k + 1] > nums[i + k])) or \
                   (pattern[k] == 0 and not (nums[i + k + 1] == nums[i + k])) or \
                   (pattern[k] == -1 and not (nums[i + k + 1] < nums[i + k])):
                    match = False
                    break
            if match:
                count += 1
        
        return count

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3036_number-of-subarrays-that-match-a-pattern-ii/solutions/java/solution.java
class Solution {
    public int countMatchingSubarrays(int[] nums, int[] pattern) {
        int count = 0;
        int m = pattern.length;
        int n = nums.length;

        // Loop through each possible starting index of a subarray of length m+1 in nums
        for (int i = 0; i <= n - m - 1; i++) {
            boolean matches = true;

            // Check each element of the pattern
            for (int k = 0; k < m; k++) {
                if ((pattern[k] == 1 && !(nums[i + k + 1] > nums[i + k])) ||
                    (pattern[k] == 0 && !(nums[i + k + 1] == nums[i + k])) ||
                    (pattern[k] == -1 && !(nums[i + k + 1] < nums[i + k]))) {
                    matches = false;
                    break;
                }
            }

            if (matches) {
                count++;
            }
        }
        return count;
    }
}
